src/System.Text.Json/src/System/Text/Json/Node/JsonArray.cs

```diff
@@ -0,0 +1,675 @@
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="ushort"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        [CLSCompliant(false)]
+        public JsonArray(IEnumerable<ushort> values) : this()
+        {
+            foreach (ushort value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="uint"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        [CLSCompliant(false)]
+        public JsonArray(IEnumerable<uint> values) : this()
+        {
+            foreach (uint value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="ulong"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        [CLSCompliant(false)]
+        public JsonArray(IEnumerable<ulong> values) : this()
+        {
+            foreach (ulong value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="decimal"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        public JsonArray(IEnumerable<decimal> values) : this()
+        {
+            foreach (decimal value in values)
+            {
+                _list.Add(new JsonNumber(value));
+            }
+        }
+
+        /// <summary>
+        ///   Gets or sets the element at the specified index.
+        /// </summary>
+        /// <param name="idx">The zero-based index of the element to get or set.</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   Index is less than 0.
+        /// </exception>
+        public JsonNode this[int idx]
+        {
+            get
+            {
+                if (idx < 0 || idx >= _list.Count)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+                return _list[idx];
+            }
+            set
+            {
+                if (idx < 0 || idx >= _list.Count)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+                _list[idx] = value;
+            }
+        }
+
+        /// <summary>
+        ///   Adds the specified <see cref="JsonNode"/> value to the JSON array.
```

Comment:
99% of the instances of "JSON" in the comments here don't really apply.  Instead of "the JSON array" here I'd say "the collection".  It may have been a JSON array value during Parse, and it may be one again during WriteTo, but for the purposes of Add it's just a collection.  You could also say "as the last item in this collection", since you're an IList (and thus ordered).

Our current docs style guides say to not really reference the current type name in method summaries.